### PR TITLE
added empty cta to playlist page + hid playlist table when empty

### PR DESCRIPTION
**Pull request by Ijin08**
fixes #12490

skarmavbild 2018-08-13 kl 16 22 07

### CODE REVIEW

File: public/app/features/playlist/partials/playlists.html [hunk 1]
```diff
@@ -1,47 +1,66 @@
...
-      </td>
-    </tr>
-  </table>
+  <div ng-if="ctrl.playlists.length === 0">
+    <empty-list-cta model="{
+      title: 'There are no playlists defined yet',
+      buttonIcon: 'fa fa-plus',
+      buttonLink: 'playlists/create',
+      buttonTitle: ' New Playlist',
+      proTip: 'ProTip: You can run the playlist in Kiosk Mode.',
```
Comment (marefr) [inline public/app/features/playlist/partials/playlists.html]: ProTip are rendered multiple times, like ` ProTip: ProTip:`. 

Something to think about for the future - the protip icon is now the same as for explore.

File: public/app/features/playlist/partials/playlists.html [hunk 2]
```diff
@@ -1,47 +1,66 @@
...
-      <td  class="text-right">
-        <a ng-click="ctrl.removePlaylist(playlist)" class="btn btn-danger btn-small">
-          <i class="fa fa-remove"></i>
-        </a>
-      </td>
-    </tr>
-  </table>
+  <div ng-if="ctrl.playlists.length === 0">
+    <empty-list-cta model="{
+      title: 'There are no playlists defined yet',
```
Comment (marefr) [inline public/app/features/playlist/partials/playlists.html]: In comparison with manage dashboards page it says "This folder doesn't have any dashboards yet". How about "There are no playlists created yet"? I'm no expert at copy, but maybe it's possible to come up with something better than defined?

File: public/app/features/playlist/partials/playlists.html [hunk 3]
```diff
@@ -1,47 +1,66 @@
...
+      buttonIcon: 'fa fa-plus',
+      buttonLink: 'playlists/create',
+      buttonTitle: ' New Playlist',
```
Comment (marefr) [inline public/app/features/playlist/partials/playlists.html]: Create Playlist? In same manner as Create Dashboard on manage dashboard page

File: public/app/features/playlist/partials/playlists.html [hunk 4]
```diff
@@ -1,47 +1,66 @@
...
-        <a ng-click="ctrl.removePlaylist(playlist)" class="btn btn-danger btn-small">
-          <i class="fa fa-remove"></i>
-        </a>
-      </td>
-    </tr>
-  </table>
+  <div ng-if="ctrl.playlists.length === 0">
+    <empty-list-cta model="{
+      title: 'There are no playlists defined yet',
+      buttonIcon: 'fa fa-plus',
```
Comment (marefr) [inline public/app/features/playlist/partials/playlists.html]: Are we planning to introduce a certain add icon for playlists? Or at least a generic add one making the button a bit bigger

**Review by marefr — CHANGES_REQUESTED** [review]
Some minor changes needed and some questions/thoughs.

File: public/app/features/playlist/partials/playlists.html (hunk 4 above)
Comment (Ijin08) [reply public/app/features/playlist/partials/playlists.html]: We probably should if everything should be consistent. But there are a couple more pages that will need the empty list cta that then also will need add icons. 
I don't know when @bulletfactory will have time to look into it though.

File: public/app/features/playlist/partials/playlists.html (hunk 1 above)
Comment (Ijin08) [reply public/app/features/playlist/partials/playlists.html]: I don't know if explore will use the rocket or a custom svg as the other icons in the side menu.
If explore is going to use the rocket we need a different icon for pro-tips.

Comment (marefr) [reply public/app/features/playlist/partials/playlists.html]: Yeah. I see that Github are using a lightbulb (at the bottom of page) for their tips.

**Review by marefr — APPROVED** [review]
LGTM :+1: 

@bulletfactory do we need to change the Create Playlist  icon to coop with the other icons used in existing CTA's, like Create Dashboard?

**Pull request merged**